e2e/config/azure.go

```diff
@@ -300,22 +334,23 @@ func (a *AzureClient) createBlobStorageAccount(ctx context.Context) error {
 }
 
 func (a *AzureClient) createBlobStorageContainer(ctx context.Context) error {
-	_, err := a.StorageContainers.Create(ctx, ResourceGroupName, Config.BlobStorageAccount(), Config.BlobContainer, armstorage.BlobContainer{}, nil)
+	_, err := a.StorageContainers.Create(ctx, ResourceGroup, Config.BlobStorageAccount(), Config.BlobContainer, armstorage.BlobContainer{}, nil)
 	if err != nil {
 		return fmt.Errorf("create blob container: %w", err)
 	}
 	return nil
 }
 
-func (a *AzureClient) assignReaderRoleToBlobStorage(ctx context.Context, principalID *string) error {
-	scope := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Storage/storageAccounts/%s", Config.SubscriptionID, ResourceGroupName, Config.BlobStorageAccount())
+func (a *AzureClient) assignRolesToVMIdentity(ctx context.Context, principalID *string) error {
```

Comment:
I only see a single Role.  are we expecting to set more roles ? I would expect the function to receive a list of roles if we want to make it more generic

Reply:
Roles are defined inside the function. I think I went through a few iterations and eventually came back to a single one.